extension/src/extension.ts

```diff
@@ -249,3 +251,41 @@ export function activate(context: vscode.ExtensionContext): void {
 export function deactivate(): void {
   // any need for cleaning?
 }
+
+// /**
+//  * Synchronizes the anonymous user ID between a local file and global workspace state, if necessary, and then starts Telemetry.
+//  * The global workspace state is preserved over vscode updates, so it is more durable than local extension file.
+//  * The user ID should then be read by the language server or debug service when sending telemetry.
+//  */
+function startTelemetry(
+  context: vscode.ExtensionContext,
+  settings: Settings
+): void {
+  let newInstallation = false;
+  let userId: string = context.globalState.get(userIdStateKey) || "";
+  try {
+    const path = context.asAbsolutePath(userIdFile);
+    if (fs.existsSync(path)) {
+      if (!userId) {
```

Comment:
Could be one condition.

Reply:
Copy&paste from AL Language extension. I think it's easier to maintain the code clone if we keep the structure ("wrong" or not 🙂) as in the original.